compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -673,10 +673,17 @@ object Types {
           pdenot.asSingleDenotation.derivedSingleDenotation(pdenot.symbol, jointInfo)
         }
         else
-          pdenot & (
+          val joint = pdenot & (
```

Comment:
I guess there's no good way to handle parameter names when intersecting two method types?

And sorry about the off-topic remark, but I'm a bit surprised to see infix with multiple parameter lists, isn't that an anti-pattern? I see it was discussed on https://github.com/lampepfl/dotty/pull/4311#issuecomment-381104279

Reply:
Yes, we should change the infix part.